### PR TITLE
add the redirectUrl argument and use it in loginUrl

### LINKED ISSUE #6: Add redirect_url query string parameter
### Description/Context
In https://github.com/mitodl/mit-open/pull/429, we added support to `mit-open` for setting a `redirect_url` query string parameter when hitting `/login/ol-oidc/`. This parameter defines a URL to redirect back to after login, granted the hostname is in `SOCIAL_AUTH_ALLOWED_REDIRECT_HOSTS`. The `mit-open-login-button` library should accept a value for this, and send it as a query string parameter when hitting the login endpoint.

### Plan/Design
 - [ ] Add a new argument to `initLoginButton` to take the redirect URL
 - [ ] Use the value from this argument as a query string parameter when hitting the `mit-open` login endpoint

### PR DESCRIPTION
**Pull request by gumaerc**
### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open-login-button/issues/6

### Description (What does it do?)
This PR adds a new argument to `initLoginButton`, `redirectUrl`. This string argument is simply passed through as the `redirect_url` query string parameter to the login endpoint. If the domain of the redirect URL is set in the `SOCIAL_AUTH_ALLOWED_REDIRECT_HOSTS` setting on the instance of `mit-open` that you are hitting, then you will be redirected back to that URL after logging in.

### How can this be tested?
 - You will need to make a slight alteration to the code before testing this, as at the time of writing https://github.com/mitodl/mit-open-login-button/pull/5 has not been merged and it fixes an issue with the `/users/me` endpoint:
```
Replace:
const apiUrl = `${parsedBaseUrl.origin}/api/v0/users/me?format=json`
With:
const apiUrl = `${parsedBaseUrl.origin}/api/v0/users/me/?format=json`
```
 - In order to test this, you will need a running instance of [`mit-open`](https://github.com/mitodl/mit-open). Follow the instructions in the `mit-open` readme for initial setup. You will need an account in your local instance for this to work. Your instance of `mit-open` should be accessible at `http://od.odl.local:8063`. You will need to set the following environment variables:
```
CORS_ALLOWED_ORIGINS=["http://ocw.odl.local:3000"]
SOCIAL_AUTH_ALLOWED_REDIRECT_HOSTS=["ocw.odl.local:3000"]
```
 - You will also need `ocw-hugo-themes` (https://github.com/mitodl/ocw-hugo-themes) set up locally, running a course using `yarn start course`. Follow the readme there for basic setup instructions and make sure you can start up a course. After you have that working, check out the `cg/add-mit-open-login-button` branch. Assuming you have cloned this repo, run `yarn add /path/to/mit-open-login-button` to add your local copy of the code for this to your locally running `ocw-hugo-themes`. This is necessary because the package is not yet published to NPM. You will also need an entry in your `hosts` file that points the `ocw.odl.local` domain to `127.0.0.1`.
 - Once you have all this running, make sure you are logged out of your local instance of `mit-open`
 - Back in your locally running OCW course site, you should see a Login button in the upper right. Click this button and log in to your locally running `mit-open`
 - After you have logged in, you will be sent back to http://ocw.odl.local:3000
 - Confirm that you see `Logged in as: username` with your username in place of `username`
 - In another tab, log out of your locally running `mit-open`
 - Back on the local OCW page, refresh and verify that the login button reappears

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,21 +4,25 @@
...
   buttonText = "Login",
   buttonClass = "",
   loggedInTextClass = "",
 ) {
   const container = document.getElementById(containerId)
   const parsedBaseUrl = new URL(baseUrl)
   const apiUrl = `${parsedBaseUrl.origin}/api/v0/users/me?format=json`
-  const loginUrl = `${parsedBaseUrl.origin}/login/ol-oidc/`
+  const redirectUrlParam =
+    redirectUrl !== "" ? `?redirect_url=${redirectUrl}` : ""
```
Comment (rhysyngsun) [inline index.js]: I think this needs to be encoded:

```suggestion
    redirectUrl !== "" ? `?redirect_url=${encodeURIComponent(redirectUrl)}` : ""
```

**Review by rhysyngsun — CHANGES_REQUESTED** [review]
Code review

File: index.js (hunk 1 above)
Comment (gumaerc) [reply index.js]: Thanks, I implemented your suggestion

**Comment by gumaerc** [comment]
@rhysyngsun This is ready for another look

**Comment by rhysyngsun** [comment]
Note: I'm not sure if there will be docs elsewhere but the instructions were missing the step to set the environment variable `MIT_OPEN_BASEURL` for the ocw-hugo-themes process.

**Comment by gumaerc** [comment]
> Note: I'm not sure if there will be docs elsewhere but the instructions were missing the step to set the environment variable `MIT_OPEN_BASEURL` for the ocw-hugo-themes process.

Yea, sorry about that. @ChristopherChudzicki merged a PR into my branch for this that adds that variable and I forgot to update the testing instructions.

**Review by rhysyngsun — APPROVED** [review]
LGTM

**Pull request merged**